api/models/release.go

```diff
@@ -671,3 +668,93 @@ func waitForTemplate(bucket string, id string) error {
 	fmt.Printf("ns=kernel at=waitForTemplate.tick error=unknown bucket=%q release=%q\n", bucket, id)
 	return fmt.Errorf("unknown error")
 }
+
+// monitorReleasePromotion observes a release's rollout by verifying the app's CF stack status
+// Sends a notifcation on success or failure. This function blocks.
+// TODO: this should be in provider.ReleasePromote()
+func waitForPromotion(r *Release) {
```

Comment:
We should get this out with the simple polling strategy to start getting some visibility into releases. We will be able to use that data to drive another iteration of release notifications.